Assets/Scripts/CartesianTiles/ImageProjectionLayer.cs

```diff
@@ -54,12 +54,16 @@ public override void HandleTile(TileChange tileChange, Action<TileChange> callba
                     tiles[tileKey].runningCoroutine = StartCoroutine(DownloadDataAndGenerateTexture(tileChange, callback));
                     break;
                 case TileAction.Upgrade:
-                    tiles[tileKey].unityLOD++;
-                    tiles[tileKey].runningCoroutine = StartCoroutine(DownloadDataAndGenerateTexture(tileChange, callback));
+                    if (tiles.ContainsKey(tileKey)) {
```

Comment:
dit gaat potentieel wel fout, want die containskey kan dus de callback stoppen, en volgens mij hangt er aan die callback weer een pending change  lijst (of andere, weet niet welke het was) die word opgeruimd. Nu kan potentieel die tilechanges lijst vol lopen. waardoor de limiet overschreden word en geen changes meer kunnen worden doorgevoerd.

Reply:
![image](https://github.com/user-attachments/assets/af644fb3-9f15-4bb6-bb5b-6e18537fa377)
Lijkt erop dat dat removen los staat van de callback